plugin/advisor/mysql/advisor_naming_column.go

```diff
@@ -0,0 +1,132 @@
+package mysql
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/pingcap/tidb/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*NamingColumnConventionAdvisor)(nil)
+)
+
+func init() {
+	advisor.Register(db.MySQL, advisor.MySQLNamingColumnConvention, &NamingColumnConventionAdvisor{})
+	advisor.Register(db.TiDB, advisor.MySQLNamingColumnConvention, &NamingColumnConventionAdvisor{})
+}
+
+// NamingColumnConventionAdvisor is the advisor checking for column naming convention.
+type NamingColumnConventionAdvisor struct {
+}
+
+// Check checks for column naming convention.
+func (adv *NamingColumnConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+	format, err := api.UnmarshalNamingRulePayloadFormat(ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	checker := &namingColumnConventionChecker{
+		level:  level,
+		format: format,
+		tables: make(tableState),
+	}
+
+	for _, stmtNode := range root {
+		(stmtNode).Accept(checker)
+	}
+
+	return checker.generateAdviceList(), nil
+}
+
+type namingColumnConventionChecker struct {
+	adviceList []advisor.Advice
+	level      advisor.Status
+	format     *regexp.Regexp
+	tables     tableState
+}
+
+func (v *namingColumnConventionChecker) Enter(in ast.Node) (ast.Node, bool) {
+	switch node := in.(type) {
+	// CREATE TABLE
+	case *ast.CreateTableStmt:
+		v.createTable(node)
+	// ALTER TABLE
+	case *ast.AlterTableStmt:
+		table := v.tables.getTable(node.Table.Name.O)
+		for _, spec := range node.Specs {
+			switch spec.Tp {
+			// RENAME COLUMN
+			case ast.AlterTableRenameColumn:
+				delete(table, spec.OldColumnName.Name.O)
+				table[spec.NewColumnName.Name.O] = true
+			// ADD COLUMNS
+			case ast.AlterTableAddColumns:
+				for _, column := range spec.NewColumns {
+					table[column.Name.Name.O] = true
+				}
+			// DROP COLUMN
+			case ast.AlterTableDropColumn:
+				delete(table, spec.OldColumnName.Name.O)
```

Comment:
So this means in one check task, we could meet `add column x`, then `drop column x`(same column), we just using the final state to do the check?

Reply:
Yes, exactly.

Reply:
Oh, I think I got it, I should just consider all the new columns instead of calculating the final state.
Let me change it.

Reply:
Updated.